hera_cal/lst_stack/metrics.py

```diff
@@ -0,0 +1,386 @@
+        for submodel, weight in zip(self.submodels[1:], self.weights[1:]):
+            cdf += submodel.cdf(x) * weight
+        return cdf
+
+    def rvs(self, size):
+        submodel_choices = np.random.choice(
+            len(self.submodels), size=size, p=self.weights
+        )
+        submodel_samples = [submodel.rvs(size=size) for submodel in self.submodels]
+        return np.choose(submodel_choices, submodel_samples)
+
+
+def zsquare_predicted_dist(df: int = 2):
+    if df in {1, 2}:
+        return chi2(df=df)
+    else:
+        raise ValueError(
+            "df should be either 1 (if using Z^2 of real/imag separately), "
+            "or 2 (if using |Z^2|)."
+        )
+
+
+@attrs.define(slots=False)
+class LSTBinStats:
+    """Class that holds basic LST-binned data and statistics"""
+    mean: DataContainer = attrs.field()
+    std: DataContainer = attrs.field()
+    nsamples: DataContainer = attrs.field()
+    flags: DataContainer = attrs.field()
+    median: DataContainer = attrs.field()
+    mad: DataContainer = attrs.field()
+    days_binned: DataContainer = attrs.field()
+
+    @classmethod
+    def from_reduced_data(
+        cls,
+        antpairs: list[tuple[int, int]],
+        pols: list[str],
+        rdc: dict[str, np.ndarray],
+        reds: RedundantGroups | None = None,
+    ):
+        """Return a class given a dictionary computed by reduce_lst_bins."""
+        outdict = {}
+        kls = partial(RedDataContainer, reds=reds) if reds is not None else DataContainer
+
+        for k, v in rdc.items():
+            if k == 'data':
+                k = 'mean'
+            dct = {antpair + (pol,): v[:, i, :, j] for i, antpair in enumerate(antpairs) for j, pol in enumerate(pols)}
+            outdict[k] = kls(dct)
+
+        return cls(**outdict)
+
+    @property
+    def bls(self):
+        return self.mean.bls()
+
+
+def get_nightly_predicted_variance(
+    bl: tuple[int, int, str],
+    stacks: list[UVData],
+    auto_stats: LSTBinStats,
+    dt: float,
+    df: float,
+) -> list[np.ndarray]:
+    """
+    Compute the predicted variance from the autos over nights/freqs for a particular bl.
+
+    Output is a list with the length of the number of lst bins, and each element is
+    an array of shape (Nnights, Nfreqs).
+    """
+
+    auto = auto_stats.mean[(bl[0], bl[0], bl[2])] * auto_stats.mean[(bl[1], bl[1], bl[2])]
+
+    per_day_expected_var = [None] * len(stacks)
+    dtdf = dt * df
+
+    for i, stack in enumerate(stacks):
+        gf = stack.get_flags(bl)
+        per_day_expected_var[i] = np.abs(auto[i] / dtdf / stack.get_nsamples(bl))
+        per_day_expected_var[i][gf] = np.inf
+
+    return per_day_expected_var
+
+
+def get_zscores(
+    auto_stats: LSTBinStats,
+    cross_stats: LSTBinStats,
+    dt: float,
+    df: float,
+    stacks: list[UVData],
+    central: str = 'mean',
+    std: str = 'autos'
+) -> list[UVData]:
+    """
+    Obtain Z-scores as a list of UVData objects.
```

Comment:
Done, thanks. I'm also updating these functions to accept only a single Stack, rather than a  list of stacks, in line with updates you suggested in `averaging.py`.